OpenRA.Game/Traits/World/Shroud.cs

```diff
@@ -383,13 +453,25 @@ public bool IsVisible(Actor a)
 
 		public bool IsTargetable(Actor a)
 		{
+			IEnumerable<CPos> vis = GetVisOrigins(a);
+			bool concealed = vis.All(c => generatedDisruptionCount[c] > 0 || generatedShroudCount[c] > 0);
```

Comment:
Although i'm still not convinced about all of this.  It sounds like you have uncovered an existing bug that must already exist in certain situations in the current code -- separating out the bugfix from your generation code would make this much easier to understand and review.